yaas/parsers/service.py

```diff
@@ -0,0 +1,23 @@
+# coding: utf-8
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import inspect
+
+from .. import service
+from .. import config
+from .. import utils
+
+def define_subcommand(parent_parser):
```

Comment:
(Maybe what I'm getting at is called Idempotency? Do you know?)

I think this example will help... Lately, I've been doing this:
``` python
def cli(parser=argparse.ArgumentParser()):
    parser.add_argument(...)
    return parser

if __name__ == '__main__':
    args = cli().parse_args()
```
instead of this
``` python
def cli_args():
    parser=argparse.ArgumentParser()
    parser.add_argument(...)
    args = parser.parse_args()
    return args

if __name__ == '__main__':
    args = cli_args()
```
because it's more flexible. It allows you to get a parser and do what you want with it (i.e. if this was a subparser, it could be standalone or vice-versa), or you could pass in your existing parser and have it added (like it is currently being used).